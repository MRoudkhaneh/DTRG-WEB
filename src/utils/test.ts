const shallowFinder = (component, attr) => component.find(`[id='${attr}']`);

export { shallowFinder };
