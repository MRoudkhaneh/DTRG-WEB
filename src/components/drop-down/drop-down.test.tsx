import { DropDown } from ".";
import { idFinder, classFinder, useTree, setUp } from "test";

describe("DropDown", () => {
  let component;
  beforeEach(
    () =>
      (component = setUp(DropDown, {
        active: true,
        children: <p className="drop-down-test">test</p>,
      }))
  );
  it("SHOUD Match <DropDownOption/> TO SNAPSHOT", () => {
    let dropDown = useTree(DropDown);
    expect(dropDown).toMatchSnapshot();
  });
  it("SHOULD RENDER WRAPPER", () => {
    const wrapper = idFinder(component, "drop-down");
    expect(wrapper.length).toBe(1);
  });
  it("SHOULD RENDER ARROW UP ICON WHEN IS ACTIVE", () => {
    const icon = idFinder(component, "arrow-up");
    expect(icon.length).toBe(1);
  });
  it("SHOULD RENDER CHILDREN", () => {
    const icon = classFinder(component, ".drop-down-test");
    expect(icon).toHaveLength(1);
  });
  //   it("SHOUD RENDER OPEN ICON BEFORE CLICK", () => {
  //     const button = idFinder(component, "button");
  //     const arrowUp = idFinder(component, "arrow-up");
  //     expect(button).toHaveLength(1);
  //     //button.simulate("click");
  //     button.props().onClick();
  //     expect(arrowUp).toHaveLength(1);
  //   });
});
