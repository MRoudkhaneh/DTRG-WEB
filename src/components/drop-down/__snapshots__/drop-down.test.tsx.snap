// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown SHOUD Match <DropDownOption/> TO SNAPSHOT 1`] = `
<div
  className="w-full  flex flex-col "
  id="drop-down"
>
  <button
    className="w-full flex justify-between items-center cursor-pointer text-gray-400 hover:text-indigo-500 focus:outline-none undefined"
    id="button"
    onClick={[Function]}
  >
    <span
      className=" text-lg"
    />
    <svg
      className=" w-5 h-5 "
      fill="currentColor"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659l4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659z"
      />
    </svg>
  </button>
  <div
    className="w-full  rounded transition-all ease-linear duration-200 pl-4 py-2   h-0"
  />
</div>
`;
